### PR TITLE
YAML + Commit Names + Files

### LINKED ISSUE #245: Set content of commits
Related to #224 
Required by #238 
Required by #235 

Various levels require files with specific contents to be present in the commits. See #238 and #235 This functionality should be added or verified to exist/work smoothly if most of the needed methods are already in place.

### PR DESCRIPTION
**Pull request by benthayer**
closes #245 

Use YAML syntax to set the commit names and to add files

### CODE REVIEW

**Comment by sahansk2** [comment]
Will you update the level templates to showcase the YAML syntax in this PR? Or will that implemented elsewhere?

**Comment by benthayer** [comment]
@sahansk2 I think this is done. Can you take a look to see if anything's missing or if any of the code doesn't make sense?

File: level_file_templates/details.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'1':
+  message: "Initial commit"
+'2':
+  message:
+    - Start solving an issue on a branch
+    - Solves issue 1
+    - Solves issue 2
+'3':
+  message:
+    - This poriton of the message will be two lines below the first line
```
Comment (sahansk2) [inline level_file_templates/details.yaml:10]: > poriton

Typo?

File: gitgud/skills/intro/_welcome/welcome.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Welcome to Git Gud, a command line game designed to help you learn how to use the popular version control system known as Git!
+
+To play the game, remember these important commands:
+
+#1) git gud goal: Shows the precise goal to complete for the current level.
+#2) git gud test: Once you've completed the current level's goals, use this to check completion.
+#3) git gud explain: Displays instructions for the current level - if you're unsure what to do next, use this!
+#4) git gud load next: Whether you've completed the current level or just want to skip it, use this to proceed to the next level!
+
+To get help with commands, run "git gud help", and for more information on Git Gud, check out (or checkout) https://github.com/benthayer/git-gud!
```
Comment (sahansk2) [inline gitgud/skills/intro/_welcome/welcome.txt:10]: Just a level design thing, but maybe we should add an explicit instruction at the end to run `git gud goal`?

File: level_file_templates/details.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - This line is below the line above
+  add-files:
+    file.txt:
+      - "'add-files' will specify file names and add content"
+      - These lines will show up in file.txt
+    another.txt:
+      - "'add-files' and 'remove-files' are meant to be used together"
+  remove-files:
```
Comment (sahansk2) [inline level_file_templates/details.yaml:18]: I'm not fully clear on what remove-files is doing in this case. Is it removing all files that were added in commits '1' and '2'?

**Comment by benthayer** [comment]
@sahansk2 can you check this out again. I updated this so files created have names such as `2.txt` instead of `2` when using `git gud commit` or setting up a level. They also come with a message. Hopefully this resolves the issues you brought up

**Comment by sahansk2** [comment]
@benthayer Yep, this all looks good!

**Pull request merged**